По нажатию на ЛК пользователь видит арендованные им инструменты

Daria-pugacheva opened the pull request:
По нажатию на ЛК пользователь видит арендованные им инструменты:
1. Добавлено ограничение, что доступ к Профилю есть только у зарегистрированных пользователей.
2. Оживила функционал загрузки данных о пользователе:
- отсмотрела актуальность html-файла
- прописаны функции в js-файле
- в UserController немного добавила метод getSelfUserInfo, по которому пользователь получает данные о себе (инфо, кто этот "себе" берется из Principal)
3. Оживила функционал добавления данных об арендованных пользователем инструментах:
 - отсмотрела актуальность html-файла
 - прописаны функции в js-файле
 - в OrderController немного поменяла входные данные в методе получения инфо по заказам + написала там предложение, как мне видится, можно это оптимизировать - коммент в строке 59. Также написала TODO-шку, что метод в OrderController по получению заказов надо сделать рабочим , а не с заглушкой, как сейчас.
 4. В таблице с инфо об арендованных инструментах добавлена кнопка "Оставить отзыв", которая перекидывать на страницу для оставления отзыва
 5. Сделала html и js наброски страницы для того, чтобы оставлять отзывы об инструменте и его владельце.

 Работоспособность, наверное, удобнее будет проверять, когда бэк будет готов. (но вообще все должно функционировать)

Daria-pugacheva committed:
По нажатию на ЛК пользователь видит арендованные им инструменты:
1. Добавлено ограничение, что доступ к Профилю есть только у зарегистрированных пользователей.
2. Оживила функционал загрузки данных о пользователе:
- отсмотрела актуальность html-файла
- прописаны функции в js-файле
- в UserController немного добавила метод getSelfUserInfo, по которому пользователь получает данные о себе (инфо, кто этот "себе" берется из Principal)
3. Оживила функционал добавления данных об арендованных пользователем инструментах:
 - отсмотрела актуальность html-файла
 - прописаны функции в js-файле
 - в OrderController немного поменяла входные данные в методе получения инфо по заказам + написала там предложение, как мне видится, можно это оптимизировать - коммент в строке 59. Также написала TODO-шку, что метод в OrderController по получению заказов надо сделать рабочим , а не с заглушкой, как сейчас.
 4. В таблице с инфо об арендованных инструментах добавлена кнопка "Оставить отзыв", которая перекидывать на страницу для отставления отзыва
 5. Сделала html и js наброски страницы для того, чтобы оставлять отзывы об инструменте и его владельце.

 Работоспособность, наверное, удобнее будет проверять, когда бэк будет готов. (но вообще все должно функционировать)

tatiananosova commented:
![image](https://user-images.githubusercontent.com/76630606/166621031-049e9bc5-7a32-42fb-9dc7-045a50121a04.png)
Пользователь не подгружается, кнопка авторизации на странице даже после авторизации.

tatiananosova commented on source/core/src/main/resources/static/profile/profile.js:
```diff
@@ -18,15 +18,15 @@ angular.module('tools').controller('profileController', function ($scope, $http)
+
+    $scope.loadMyProfile = function () {
+        $http({
+            url: contextPath + '/api/v1/users',
```
Мне кажется, что этот эндпоинт возвращает всех пользователей, а нам нужен конкретный.

Daria-pugacheva replied:
Про подгрузку пользователя посмотрю, почему так. А что касается энд-поинта, то он возвращает строго одного юзера, имя которого берет из Principal (я специально метод в UserController добавила)

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-25

Daria-pugacheva committed:
Доработала, чтобы разные div-ы показывлись в зависимости от того, авторизован пользователь или нет.
Что сделано:
1. В ЛК есть доступ только, если юзер авторизован. Если не авторизован, то он видит соответствующую надпись и может тут же воспользоваться кнопкой Авторизваться - эта кнопка перебрасывает на страницу авторизации.
2. После авторизации в ЛК уже видно личные данные пользователя (поправлен запрос на бэк - добавлен энд-поинт для нового метода UserController-а)
3. Таблица со списком арендованных инструментов пока заполнена не данными с бэка (пока не доработано на бэке), а тестовыми данными и видно, что можно нажать на кнопку "Оставить отзыв" по нажатию на которую мы перходим на страницу отзывов. Пока при переходе на страницу отзывов выскакивает alert, т.к. данные об инструменте таком подгрузить не удалось (ибо инструмент тестовый, а функция ждет id инструмента из загруженного списка, которого пока нет). Но саму страницу при этом можно посмотреть, пока не нажали OK  на alert-е

И вот что я обнаружила: если оставить незакомментированной фукнцию запроса заказов с фронта, то ответом на нее прилетает ошибка (т.к. бэк не готов) и ПРИ ЭТОМ сбивается ограничение фронта в части показа тех div-ов ЛК, которые связаны с флагом авторизован/не авторизован - появляются оба div-а... В упор не понимаю пока, почему такая реакция.
При этом я вот эту функцию загрузки заказов закомментировала, пока она не работает, и все функционирует нормально...

siporqueno committed:
Merge pull request #52 from siporqueno/LMR-49_59

Added backend necessary for this branch

siporqueno commented:
Вмержил в LMR-25 изменения сделанные Дмитрием в LMR-49_59. Эндпойнт api/v1/instruments/lk дает список своих инструментов и теперь можно все, что связано с выводом своих инструментов в лк раскомментировать и посмотреть как работает.

Daria-pugacheva committed:
1. Оживила функционал отражения таблицы со своими инструментами, размещенными на сайте. Убрала из этой таблицы столбцы про арендатора и даты начала и окончания аренды, т.к. теперь на подробную инфо по арендам переходим по кнопке (это реализовано пока в отдельной ветке № 43, а не здесь).
2. В InstrumentController написала TODO, чтобы удаление инструмента было доступно не только админу, но и владельцу инструмента. Если нет возражений по меой логике, то надо будет поправить метод контроллера.
3. Заодно непножко подаравила энд-поинты в edit-tool.js
Пока изменения только сделала в коде, приложение смогу запустить только во вторник, чтобы посмотреть, как по факту работает.

siporqueno commented on source/core/src/main/java/ru/letmerent/core/controllers/OrderController.java:
```diff
@@ -65,8 +65,8 @@ ResponseEntity<OrderDto> getOrderById(@Parameter(description = "Идентифи
             ))
-    ResponseEntity<Collection<OrderDto>> getOrdersByUserId(@RequestParam Long userId) {
-        return new ResponseEntity<>(Collections.emptyList(), HttpStatus.OK);
+    ResponseEntity<Collection<OrderDto>> getOrdersByUserId(@PathVariable Long userId) {
```
```suggestion
    ResponseEntity<Collection<OrderDto>> getOrdersByUserId(@RequestParam Long userId) {
```

siporqueno replied:
Здесь нужно на мой взгляд оставить @RequestParam, иначе эндпойнт GET orders/{id} будет конфликтовать с GET  orders/{userId}. Пусть он в параметре передается, тогда не будет конфликта.

Daria-pugacheva replied:
Мысль поняла и согласна. Смущает в твоем предложении то, что сейчас метод-то у нас GET, т.е. RequestParam - это концептуально не про него, т.е. надо и тип обмена менять.  Пока, давай, я править не буду: там этот метод все равно надо дорабатывать, т.к. в нем заглушка.  Обозначим тогда Алексею этот момент на принятие решения и когда он сделает рабочий бэк по отдаче заказов по айдишнику юзера, тогда я фронт быстренько скорректирую.  В том числе я изначально в этом методе оставила еще комментарий-предложение, чтобы не передавать никакие айдишники, а искать заказ по имени пользователя из Principal, т.е. просто сделать отдельный энд-поинт типа /myOrders . 
Я в TODO-шке допишу, чтобы Алексей обратил внимание на энд-поинт в том числе в том ключе, что мы с тобой проговорили. (Правда, чтобы он ее увидел, надо будет вмержить ветку в девелоп. Я написала Татьяне просьбу, чтобы смотрела-одобряла)

siporqueno reviewed (approved):
По поводу эндпойнта для заказов пользователя оставил комментарий.

Daria-pugacheva committed:
Merge pull request #58 from siporqueno/LMR-4

LMR-4 - В ЛК пользователь видит список своих инструментов, выставленных на сайте для передачи в аренду

Daria-pugacheva committed:
1. Кнопку "Удалить" оставила для дописания функционала запроса администратору на удаление.
2. Добавила кнопку снять с аренды с комментарием в методе.
Надо будет обсудить, в каком ключе дорабатываем этот функционал - все есть в комментариях в profile.js (строки 36-37 и 41-45). Мне видится, что надо бэк докручивать, а на фронте только энд-поинты останется указать нужные
3. В OrderController обновила TODO для доработки Алексеем. Когда будет готов бэк, проправим энд-поинты на фронте для загрузки заказов

Daria-pugacheva committed:
Merge branch 'develop' into LMR-25
